code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/projects/ProjectResourceManager.java

```diff
@@ -135,22 +135,28 @@ protected Object createInternal(String type, ApiRequest request) {
         if (!ProjectConstants.TYPE.equals(type)) {
             return null;
         }
-        return createProject(request);
+        return createProject(type, request);
     }
 
     @SuppressWarnings("unchecked")
-    private Account createProject(ApiRequest apiRequest) {
+    private Account createProject(String type, ApiRequest apiRequest) {
         Policy policy = (Policy) ApiContext.getContext().getPolicy();
         Map<String, Object> project = CollectionUtils.toMap(apiRequest.getRequestObject());
         if (authDao.getAccountById(policy.getAccountId()).getKind().equalsIgnoreCase(ProjectConstants.TYPE)) {
             throw new ClientVisibleException(ResponseCodes.FORBIDDEN);
         }
-        Account newProject = authDao.createProject((String) project.get("name"), (String) project.get("description"));
-        List<Map<String, String>> members = (ArrayList<Map<String, String>>) project.get("members");
-        projectMemberResourceManager.setMembers(newProject, members);
-        policy.grantObjectAccess(newProject);
-
-        return newProject;
+        Object object = super.createInternal(type, apiRequest);
+        if (object instanceof Account) {
+            Account newProject = (Account) object;
+            newProject.setKind(AccountConstants.PROJECT_KIND);
+            objectManager.persist(newProject);
```

Comment:
sorry, missed that. thought line 152 does the object create, but it just updates the kind